src/main/groovy/com/bmuschko/gradle/clover/CloverPlugin.groovy

```diff
@@ -206,11 +208,13 @@ class CloverPlugin implements Plugin<Project> {
      * @param task Task
      */
     private void setCloverReportConventionMappings(Project project, CloverPluginConvention cloverPluginConvention, Task task) {
-        task.conventionMapping.map('reportsDir') { new File(project.buildDir, 'reports') }
+        def reportDir = new File(project.buildDir, 'reports')
+        task.conventionMapping.map('reportsDir') { reportDir }
         task.conventionMapping.map('xml') { cloverPluginConvention.report.xml }
         task.conventionMapping.map('json') { cloverPluginConvention.report.json }
         task.conventionMapping.map('html') { cloverPluginConvention.report.html }
         task.conventionMapping.map('pdf') { cloverPluginConvention.report.pdf }
+        task.outputs.dir(new File(reportDir, 'clover'))
```

Comment:
We needed to add this dependency because we were experiencing a situation where reports would generate, we would do a clean, and then when we'd rerun the cloverGenerateReport task.  When this happened, there were other tasks writing to the report directory, and, when the cloverGenerateReport task finally ran, gradle thought it was up-to-date, so, the report was never created.  Adding an explicit dependency of reportDir/clover fixed the problem.  I have tried my best to add an integration test for this situation, but, I don't seem to be able to reproduce the bug outside of my actual build environment.  If you have suggestions, I'd gladly implement them.